app.js

```diff
@@ -249,44 +277,292 @@ const app = Vue.createApp({
+                eq3eq6_next_lower_macro_right = "".concat(eq1_val,eq2_val,eq3_val+1,eq4_val,eq5_val,eq6_val)
+            }
+            else{
+                //21 --> 32 (do not exist)
+                eq3eq6_next_lower_macro = "".concat(eq1_val,eq2_val,eq3_val+1,eq4_val,eq5_val,eq6_val+1)
+            }
+
 
-            // EQ1 Min Score Differential
-            value = parseFloat(value)
+            eq4_next_lower_macro = "".concat(eq1_val,eq2_val,eq3_val,eq4_val+1,eq5_val,eq6_val)
+            eq5_next_lower_macro = "".concat(eq1_val,eq2_val,eq3_val,eq4_val,eq5_val+1,eq6_val)
 
-            AV_diff={"N": 0.3, "A": 0.2, "L": 0.1, "P": 0}
-            PR_diff={"N": 0.2, "L": 0.1, "H": 0}
-            UI_diff={"N": 0.2, "P": 0.1, "A": 0}
 
-            if(lookup[0] == "0") {
-                value = value
+            //get their score, if the next lower macro score do not exist the result is NaN
+            score_eq1_next_lower_macro = lookuptable[eq1_next_lower_macro]
+            score_eq2_next_lower_macro = lookuptable[eq2_next_lower_macro]
+
+            if (eq3==0 && eq6==0){
+                //multiple path take the one with higher score
+                score_eq3eq6_next_lower_macro_left = lookuptable[eq3eq6_next_lower_macro_left]
+                score_eq3eq6_next_lower_macro_right = lookuptable[eq3eq6_next_lower_macro_right]
+
+                if (score_eq3eq6_next_lower_macro_left>score_eq3eq6_next_lower_macro_right){
+                    score_eq3eq6_next_lower_macro = score_eq3eq6_next_lower_macro_left
+                }
+                else{
+                    score_eq3eq6_next_lower_macro = score_eq3eq6_next_lower_macro_right
+                }
             }
-            else if(lookup[0] == "1") {
-                value_0 = parseFloat(this.cvssLookupData["0" + lookup.slice(1)])
-                value = Math.min(value_0, value + AV_diff[this.m("AV")] + PR_diff[this.m("PR")] + UI_diff[this.m("UI")] - 0.2)
+            else{
+                score_eq3eq6_next_lower_macro = lookuptable[eq3eq6_next_lower_macro]
             }
-            else if(lookup[0] == "2") {
-                value_1 = parseFloat(this.cvssLookupData["1" + lookup.slice(1)])
-                value = Math.min(value_1, value + AV_diff[this.m("AV")] + PR_diff[this.m("PR")] + UI_diff[this.m("UI")])
+
+
+            score_eq4_next_lower_macro = lookuptable[eq4_next_lower_macro]
+            score_eq5_next_lower_macro = lookuptable[eq5_next_lower_macro]
+
+            //get all max vector for the eq
+            eq1_maxes = this.getvalueEqLookup(lookup,0)
+            eq2_maxes = this.getvalueEqLookup(lookup,1)
+            eq3_eq6_maxes = this.getvalueEqLookup(lookup,2)[lookup[5]]
+            eq4_maxes = this.getvalueEqLookup(lookup,3)
+            eq5_maxes = this.getvalueEqLookup(lookup,4)
+
+            //compose them
+            max_vectors = []
+            for (eq1_max of eq1_maxes){
```

Comment:
Are 5 nested loops necessary for this algorithm? It is hard to judge without proper computational requirements. It looks like Cartesian product of the arrays, so maybe we should code it properly?